pandas/core/frame.py

```diff
@@ -3830,7 +3839,7 @@ def _getitem_multilevel(self, key):
             else:
                 new_values = self.values[:, loc]
                 result = self._constructor(
-                    new_values, index=self.index, columns=result_columns
+                    new_values, index=self.index, columns=result_columns, copy=False
```

Comment:
Are we sure here? `loc` can be a slice, in which case `self.values[:, loc]` can be a view?

Reply:
Yeah you are correct, this was wrong before as well. Will open a separate pr.

Reply:
Don't we generally _want_ a view?

Reply:
Yes, but we should track references when we get a view. Which we did not do before

Reply:
This was handled by https://github.com/pandas-dev/pandas/pull/51944

Reply:
Yep